src/System.Net.Http.WinHttpHandler/tests/FakeMarshal.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using WinHttpHandlerTests;
+
+namespace System.Net.Http
+{
+    public static class Marshal
+    {
+        public static int GetLastWin32Error()
+        {
+            if (TestControl.LastWin32Error != 0)
+            {
+                return TestControl.LastWin32Error;
+            }
+
+            return System.Runtime.InteropServices.Marshal.GetLastWin32Error();
+        }
+
+        public static IntPtr AllocHGlobal(int cb)
+        {
+            return System.Runtime.InteropServices.Marshal.AllocHGlobal(cb);
+        }
+
+        public static void FreeHGlobal(IntPtr hglobal)
+        {
+            System.Runtime.InteropServices.Marshal.FreeHGlobal(hglobal);
+        }
+
+        public static string PtrToStringUni(IntPtr ptr)
+        {
+            return System.Runtime.InteropServices.Marshal.PtrToStringUni(ptr);
+        }
+
+        public static IntPtr StringToHGlobalUni(string s)
+        {
+            return System.Runtime.InteropServices.Marshal.StringToHGlobalUni(s);
+        }
+
+        public static void Copy(IntPtr source, byte[] destination, int startIndex, int length)
+        {
+            System.Runtime.InteropServices.Marshal.Copy(source, destination, startIndex, length);
+        }
+
+        public static int SizeOf<T>()
+        {
+            return System.Runtime.InteropServices.Marshal.SizeOf<T>();
+        }
+
+        public static IntPtr UnsafeAddrOfPinnedArrayElement<T>(T[] arr, int index)
+        {
+            return System.Runtime.InteropServices.Marshal.UnsafeAddrOfPinnedArrayElement<T>(arr, index);
+        }
+        
+        //public static IntPtr UnsafeAddrOfPinnedArrayElement(Array arr, int index)
+        //{
+        //    return System.Runtime.InteropServices.Marshal.UnsafeAddrOfPinnedArrayElement(arr, index);
+        //}
+
+        public static T PtrToStructure<T>(IntPtr ptr)
+        {
+            return System.Runtime.InteropServices.Marshal.PtrToStructure<T>(ptr);
+        }
+
+        public static void WriteByte(IntPtr ptr, int ofs, byte val)
+        {
+            System.Runtime.InteropServices.Marshal.WriteByte(ptr, ofs, val);
+        }
+
+        public static string PtrToStringAnsi(IntPtr ptr, int len)
+        {
+            return System.Runtime.InteropServices.Marshal.PtrToStringAnsi(ptr, len);
+        }
+
+        public static void StructureToPtr<T>(T structure, IntPtr ptr, bool fDeleteOld)
+        {
+            System.Runtime.InteropServices.Marshal.StructureToPtr<T>(structure, ptr, fDeleteOld);
+        }
+        
+        //public static void StructureToPtr(object structure, IntPtr ptr, bool fDeleteOld)
```

Comment:
@davidsh Are these and other commented functions in this file still needed at a later point? If yes, maybe they could be prepended by `//TODO`.